### DISCUSSION TITLE
Feature Request: New API Additions (Round 2)

### DISCUSSION THREAD

**Discussion by MattTheCuber** [HypixelDev/PublicAPI#572] [Ideas]
Good morning Hypixel Developers,

This is a follow-up of #542 .

First off, I just wanted to thank the developers for all that they do for their players and us. By creating and updating your API, you allow external developers to build amazing tools for players to benefit by. I work with the SkyHelper development team and have been using your API for several years. This post is a list of many of the missing features that many developers have requested. I have trimmed up the list as best I could to remove features that aren't critical. We completely understand if some of these are to complicated to be added to the API. However, we plead that you look through the list and add everything that you can. Thanks for your time and I hope this post finds you feeling generous to the SkyBlock community!

**Most Important Features**
- ~~**Museum**~~ Added 👍
The museum has been out for over a year now and players still don't have a way to view their museum external from in-game. This is hands down the most requested feature for most Hypixel SkyBlock applications.
- ~~**SkyBlock Level Experience**~~ Added 👍 
We created #565 a little while ago asking about the missing information for calculating SkyBlock Levels. In that post @LandonDev said that SkyBlock Level Experience would be added to the API soon.
- **Personal Bank**
This feature was added in the last feature request post. However, soon after it was removed because of #544 . It would be really cool if this could be added back with proper API disabling.
- **Garden Level**
This was recently added to game and it would be great to have a bunch of data such as garden level experience, crop upgrade levels, crop milestones, and visitor data.
- **Jacob's Contests**
Not sure the best way to implement this, but Jacob's farming contests are a highly requested feature that the community could benefit from seeing in the API.

**Useful Features**
- ~~**Crystal Nucleus Runs**~~ Added 👍 
This would probably be very easy to add to the API and would help in manual SkyBlock level calculations.
- ~~**Bank Upgrade Level**~~ Added 👍 
This would probably also be very easy to add to the API and would help in manual SkyBlock level calculations.
- ~~**Rift Progression and Inventories**~~ Added 👍
Requested by players
- **Sack of Sacks**
We have sack contents, but it would be cool to know what sacks they own and how many pocket in a sack are applied.
- **Profile's Kat**
It would be very cool to see what pet and how much time is left for a player profile's current Kat upgrade (if in use).
- **Profile Commissions**
Just the number of commissions completed in the Dwarven Mines and the number of commissions completed in the Crystal Hollows
- **City Project Buffs**
Very simple just a list of city project and what bonus the player received in that project. Mostly useful for finding discounts on shops and slayer items.
- **Profile Dungeon Secrets**
We already have this in the achievements portion of the `/player` endpoint. However, this counts all profiles combined. It would be cool to find the secrets per profile in the `/skyblock/profiles` endpoint.
- **Perkpocalyse Mayor Order**
It would be amazing if the `/skyblock/mayor` endpoint showed the order of Perkpocalyse mayors when Mayor Jerry is elected.
- `buyer_profile` **in** `skyblock/auctions_ended`
See #545 
- **Booster Cookie and God Potion** This might have been rejected in the past?
Not critical, but, it would be cool to check if a player has an active Booster Cookie and God Potion and perhaps when they will expire.

**Island Data** You can ignore this section completely if it is too complex internally.
- **Placed Minions**
Number and types of placed minions. Possibly also the inventory of those minions such as upgrades and fuel.
- **Island Beacon**
Placed Island Beacon and current configuration.

**Comment by Altpapier** [comment]
👍

**Reply by Eejit43** [reply]
Also: Traveling Zoo data!

**Reply by MattTheCuber** [reply]
> Also: Traveling Zoo data!

Not was critical as most features on this list. Maybe next list XD

**Reply by scoutskylar** [reply]
Traveling Zoo data would still be pretty nice to have.

**Reply by MattTheCuber** [reply]
For what exactly? I don't really know many uses

**Reply by scoutskylar** [reply]
I guess now that Oringo can be added to the Abiphone, it wouldn't be as useful, but for players that don't have an Abiphone yet, it would be nice to be able to see what pets Oringo is selling with a mod or website that uses the API. I do agree that the other things listed here are more critical, though.

**Comment by auradoescoding** [comment]
W feature request

**Comment by Baltrazx** [comment]
👍

**Comment by DuckySoLucky** [comment]
👍

**Comment by 0ri4** [comment]
👍

**Comment by UP929312** [comment]
Agreed

**Comment by RenamQ** [comment]
Oh yea we need this

**Comment by Conutik** [comment]
yes

**Comment by noemtdev** [comment]
👍

**Comment by bishoperic** [comment]
I am pretty certain it's been said they won't give out the booster cookie duration, but god pots should be pretty easily calculable from the `active_effects` field.
Also from what I've understood previously about island data, everything is stored in Hypixel's .slime files, and they are not willing to add additional processing to api data. They will only expose fields in the api that they already have internally. If they've changed their minds on things like this though, I would welcome the extra info.
Other than that though, everything here looks great 👍

**Reply by MattTheCuber** [reply]
Thanks for the information. Any chance you could find references to these statements? I'll remove the Booster Cookie and God Pot feature request if so. Also, I'll add a note that they can ignore the island data stuff if they want.

**Comment by TGWaffles** [comment]
Fantastic request.

Would also like to echo 1000x the request for the Museum API - it's the biggest missing feature imo as items can just "disappear" if they're in the museum. Would be fantastic if this was added!

Also would love the `buyer_profile` in auctions_ended.

Aside from that, all fantastic suggestions & everything suggested here are good additions to the already fantastic API.

**Reply by Altpapier** [reply]
It's a pain that museum is not added yet 😭

**Reply by benc1ark** [reply]
Agreed

**Comment by axlecoffee** [comment]
👍

**Comment by wekul** [comment]
Along with the booster cookie/god potion, would be nice to be able to see how many bits a player has on the profile too.

**Reply by MattTheCuber** [reply]
That was denied in #542

**Reply by wekul** [reply]
Damn that's unfortunate

**Comment by ghost** [comment]
👍

**Reply by ghost** [reply]
Asked my friend with connections with admins to send it to Jaya

**Comment by Rigner** [comment]
I already asked for a museum API a long time ago (february 2022), Connor said it wasn't planned. Hope it changes tho, I really want it too.

**Comment by ghost** [comment]
SMP Api for the following things:
Player Count
Which players are on?
A dynmap(optional)
SMP chat -> Discord Bridge
I told this as we are discussing about Hypixel API and not exactly Skyblock Api

**Comment by MattTheCuber** [comment]
Thanks for adding skyblock experience! Greatly apricated!

**Reply by axlecoffee** [reply]
They were probably waiting for skyblock levels V2 (which rolled out around 2 hours ago)

**Comment by MattTheCuber** [comment]
Again, thank you for all that you do for us! SkyBlock Level is now almost fully calculatable using the new data from the `leveling` key:
![image](https://user-images.githubusercontent.com/32849887/217563411-82dea17a-0b63-4e3f-8ddf-1474c75fadad.png)
This is much appreciated by all SkyBlock developers! We look forward to future additions and wish the Hypixel Staff good luck in making the proper decisions for the future of the API.
